File: backend/app/schemas/liquidation.py
from typing import Literal

from app.schemas.base import CamelModel

LiquidationSide = Literal["LONG", "SHORT"]


class LiquidationEvent(CamelModel):
    id: str
    symbol: str
    side: LiquidationSide
    amount_usd: float
    price: float
    exchange: str
    timestamp: str


class LiquidationHeatmapCell(CamelModel):
    price: float
    intensity: float


class LiquidationTotals(CamelModel):
    long_usd: float
    short_usd: float


class LiquidationUpdate(CamelModel):
    """Live push over the `liquidation` WebSocket channel — see
    `app/tasks/live_feed.py`'s `_handle_liquidation`."""

    symbol: str
    side: LiquidationSide
    amount_usd: float
    price: float
    exchange: str
    timestamp: int

File: types/liquidation.ts
import type { Direction } from "./market";

export interface LiquidationEvent {
  id: string;
  symbol: string;
  side: Extract<Direction, "LONG" | "SHORT">;
  amountUsd: number;
  price: number;
  exchange: string;
  timestamp: string;
}

export interface LiquidationHeatmapCell {
  price: number;
  intensity: number; // 0-1
}

export interface LiquidationTotals {
  longUsd: number;
  shortUsd: number;
}
